data_steward/tools/create_update_drc_id_match_table.py

```diff
@@ -0,0 +1,229 @@
+
+    :param client: bq client
+    :param project_id: the project containing the dataset
+    :param table_id: ID of the table
+    """
+
+    fields = resources.fields_for(IDENTITY_MATCH_TABLE)
+
+    create_table = CREATE_TABLE.render(project_id=project_id,
+                                       drc_dataset_id=DRC_OPS,
+                                       id_match_table_id=table_id,
+                                       fields=bq.get_bq_fields_sql(fields))
+    client.query(create_table)
+
+    LOGGER.info(f'Created {table_id}.')
+
+    return table_id
+
+
+def copy_ps_values_data_to_id_match_table(client, project_id, table_id, hpo_id):
+    """
+    Copies data from the ps_values table to the DRC identity match table.
+
+    :param client: bq client
+    :param project_id: the project containing the dataset
+    :param table_id: ID for the table
+    :param hpo_id: ID for the HPO site
+    """
+
+    schema_list = bq.get_table_schema(IDENTITY_MATCH_TABLE)
+    id_match_table_id = table_id
+    ps_values_table_id = f'{PS_API_VALUES}_{hpo_id}'
+
+    fields_name_str = ',\n'.join([item.name for item in schema_list])
+
+    insert_query = INSERT_QUERY.render(project_id=project_id,
+                                       drc_dataset_id=DRC_OPS,
+                                       id_match_table_id=id_match_table_id,
+                                       fields=fields_name_str,
+                                       ps_values_table_id=ps_values_table_id)
+
+    client.query(insert_query)
+
+    LOGGER.info(
+        f'Inserted data from `{ps_values_table_id}` to `{id_match_table_id}`')
+
+
+def get_set_expression():
+    """
+    This method generates the SET_EXPRESSION query
+    """
+    set_expression = []
+    schema_list = bq.get_table_schema(IDENTITY_MATCH_TABLE)
+    ', '.join([item.name for item in schema_list])
+
+    for item in schema_list:
+        set_expression.append(SET_EXPRESSION.render(col=item.name))
+
+    # Work around to remove person_id from the fields list since person_id is required
+    # and will never need to be updated to missing_rdr
+    set_expression.remove("\nperson_id = 'missing_rdr'")
+
+    return ', '.join(set_expression)
+
+
+def update_site_drc_table(client, project_id, table_id):
+    """
+    Updates the site's DRC identity match table. If the record is empty, each field should default
+        to `missing_rdr`, else `missing_ehr` for each field that does not have any data.
+
+    :param client: bq client
+    :param project_id: the project containing the dataset
+    :param table_id: ID of the table
+    """
+    schema_list = bq.get_table_schema(IDENTITY_MATCH_TABLE)
+    fields_name_str = ',\n'.join([item.name for item in schema_list])
+
+    # person_id will not need to be updated since it will always be in table since it
+    # is a required field
+    fields = fields_name_str.replace('person_id,', '')
+    id_match_table_id = table_id
+
+    # Updates missing rows to contain 'missing_rdr'
```

Comment:
From my understanding of the ticket, I thought missing column values, not entire rows, should be replaced with "missing_rdr". Is that true?

Reply:
This is what the ticket says at the end of the Scope paragraph:
> The record for each field should default to missing_rdr if the joined record in the ps_api_values_<hpo_id> table does not contain any information.  Otherwise, default to missing_ehr

I'm interpreting that as, if there is no other value in the record (which is a row right?) than the `person_id` since that is a required field, it should say `missing_rdr` because that would mean that we didn't get any information from the RDR participant summary API.

Reply:
I see, it could be that. But what if all fields are available except one. @lrwb-aou can you weigh in?

Reply:
@jp3477 and @dcampbell-vumc Good questions!  I had to review the ticket to see what is going on.  I think @jp3477's interpretation is correct based on the third acceptance criteria and the commented conversation with Caroline.  

> The record defaults to missing_rdr if no RDR data exists for that person **in that field**.  Otherwise defaults to missing_ehr.

The premise is that for each field we have two possible default values and the default value chosen is based on the "importance" of the missing data.  If we are missing the RDR data, we can't compare anything that is submitted for that field to anything, because we just don't have the data.  

For example, a participant doesn't provide their middle name through the survey portal but their middle name is in their ehr record.  This should record the default value for the field "middle_name" as "missing_rdr" to indicate we can't compare the ehr record data for middle name to the rdr record data for middle name because we are "missing_rdr" data for this field for this participant.

This is in contrast to having middle name data provided by the participant through survey data and not having middle name information available from ehr records.  In this case, the default value would be "missing_ehr" because we can't compare the two values since we are missing the ehr data for this participant for this field.

In the case where middle name data is missing from both ehr and rdr data, we would record the field as "missing_rdr".  Survey data is considered more reliable as it comes directly from the participant.

But maybe, this is not what the question is about.  The default value should be set for each field in the record.  The presence or absence of first name data should not affect the default value of middle name data.  I think the quoted above should be changed to 

> Each field should default to missing_rdr if the joined record in the ps_api_values_<hpo_id> table does not contain any information. Otherwise, default to missing_ehr

The default value should be set based on the data available for each field in the participant record.  It is improbable that we will be missing an entire row of data from the survey/rdr side.  It is likely that we will be missing some fields.

We are interested in defaulting field values, not entire records.

Reply:
Got it, that makes sense.